service/awsconfig/v5/resource/cloudformation/adapter/adapter.go

```diff
@@ -66,7 +67,14 @@ type Config struct {
 }
 
 func NewGuest(cfg Config) (Adapter, error) {
-	a := Adapter{}
+	a := Adapter{
+		Outputs: &outputsAdapter{},
+	}
+
+	err := a.Outputs.Adapt(cfg)
```

Comment:
This Adapt method confsuses me. I thought it's used to mutate the struct. But it's used to construct the struct. It looks like a constructor of `outputsAdapter`, so why not just regular `new` function?

Reply:
It is constructed above with `&outputsAdapter{}`. Here the constructed instance is adapted.

Reply:
I saw that already and I think it's confusing, as it can be replaced by `outputs, err := newOutputsAdapter(cfg)` which is kind of idiomatic construction. So I don't follow why `Adapt`?

Reply:
No strong reason. I can change this to whatever does not confuse you. I do not mind.

Reply:
I moved the output adapter back to the loop. The signature stays. We just do not call it "manually".